salt/utils/pkg/win.py

```diff
@@ -0,0 +1,1295 @@
+        return self
+
+    def __next__(self):
+        """Returns next Package Id.
+
+        Returns:
+            str: Package Id
+
+        """
+        try:
+            return self.__iter_list.popleft()
+        except IndexError:
+            self.__iter_list = None
+            raise StopIteration
+
+    def next(self):
+        """Returns next Package Id.
+
+        Returns:
+            str: Package Id
+
+        """
+        return self.__next__()
+
+    def get(self, pkg_id, default_value=None):
+        """Returns information on a package.
+
+        Args:
+            pkg_id (str): Package Id of the software/component.
+            default_value: Value to return when the Package Id is not found.
+
+        Returns:
+            dict or list: List if ``version_only`` is ``True`` otherwise dict
+
+        """
+        return self.__reg_software.get(pkg_id, default_value)
+
+    @staticmethod
+    def __oldest_to_latest_version(ver1, ver2):
+        '''Used for sorting version numbers oldest to latest
+        '''
+        return 1 if LooseVersion(ver1) > LooseVersion(ver2) else -1
+
+    @staticmethod
+    def __latest_to_oldest_version(ver1, ver2):
+        ''''Used for sorting version numbers, latest to oldest
+        '''
+        return 1 if LooseVersion(ver1) < LooseVersion(ver2) else -1
+
+    def pkg_version_list(self, pkg_id):
+        """Returns information on a package.
+
+        Args:
+            pkg_id (str): Package Id of the software/component.
+
+        Returns:
+            list: List of version numbers installed.
+
+        """
+        pkg_data = self.__reg_software.get(pkg_id, None)
+        if not pkg_data:
+            return []
+
+        if isinstance(pkg_data, list):
+            # raw data is 'pkgid': [sorted version list]
+            return pkg_data  # already sorted oldest to newest
+
+        # Must be a dict or OrderDict, and contain full details
+        installed_versions = list(pkg_data.get('version').keys())
+        return sorted(installed_versions, key=cmp_to_key(self.__oldest_to_latest_version))
+
+    def pkg_version_latest(self, pkg_id):
+        """Returns a package latest version installed out of all the version
+        currently installed.
+
+        Args:
+            pkg_id (str): Package Id of the software/component.
+
+        Returns:
+            str: Latest/Newest version number installed.
+
+        """
+        return self.pkg_version_list(pkg_id)[-1]
+
+    def pkg_version_oldest(self, pkg_id):
+        """Returns a package oldest version installed out of all the version
+        currently installed.
+
+        Args:
+            pkg_id (str): Package Id of the software/component.
+
+        Returns:
+            str: Oldest version number installed.
+
+        """
+        return self.pkg_version_list(pkg_id)[0]
+
+    @staticmethod
+    def __sid_to_username(uid):
+        """Provided with a valid Windows User SID and returns a Username
```

Comment:
Provided with a valid Windows User Sid (uid), return a Username


Consider using/improving the `get_name` function from `salt.utils.win_dacl`
https://github.com/saltstack/salt/blob/develop/salt/utils/win_dacl.py#L1097